application_sandbox/external_memory_host/main.cpp

```diff
@@ -0,0 +1,505 @@
+#include "cube.frag.spv"
+    ;
+
+struct ExternalMemoryHostFrameData {
+  containers::unique_ptr<vulkan::VkCommandBuffer> command_buffer_;
+  containers::unique_ptr<vulkan::VkFramebuffer> framebuffer_;
+  containers::unique_ptr<vulkan::DescriptorSet> cube_descriptor_set_;
+};
+
+#ifdef _WIN32
+#include <malloc.h>
+void* aligned_alloc(size_t alignment, size_t size) {
+  return _aligned_malloc(size, alignment);
+}
+#endif
+
+void aligned_free(void* pMemory) {
+#ifdef _WIN32
+  _aligned_free(pMemory);
+#else
+  free(pMemory);
+#endif
+}
+
+template <typename T>
+class VkBufferExternalMemoryHost {
+ public:
+  VkBufferExternalMemoryHost(vulkan::VulkanApplication* application,
+                             logging::Logger* log, size_t num_images,
+                             VkBufferUsageFlagBits usage)
+      : device_(application->device()),
+        log_(log),
+        buffer_(VK_NULL_HANDLE, nullptr, &application->device()),
+        device_memory_(VK_NULL_HANDLE, nullptr, &application->device()),
+        aligned_data_size_(
+            vulkan::RoundUp(sizeof(T), vulkan::kMaxOffsetAlignment)) {
+    VkBuffer buffer;
+
+    VkBufferCreateInfo create_info = {
+        VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO,  // sType
+        nullptr,                               // pNext
+        0,                                     // flags
+        aligned_data_size_ * num_images,       // size
+        VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT |
+            VK_BUFFER_USAGE_TRANSFER_DST_BIT,  // usage
+        VK_SHARING_MODE_EXCLUSIVE,
+        0,
+        nullptr};
+
+    LOG_ASSERT(
+        ==, log, VK_SUCCESS,
+        device_->vkCreateBuffer(device_, &create_info, nullptr, &buffer));
+
+    buffer_.initialize(buffer);
+
+    VkMemoryRequirements requirements;
+    device_->vkGetBufferMemoryRequirements(device_, buffer, &requirements);
+    allocation_size_ = vulkan::RoundUp(requirements.size, 4096);
+
+    VkPhysicalDeviceExternalMemoryHostPropertiesEXT
+        physical_device_external_memory_host_properties{
+            VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTERNAL_MEMORY_HOST_PROPERTIES_EXT,
+            nullptr,
+            0,
+        };
+    VkPhysicalDeviceProperties2 physical_device_properties2{
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2,
+        &physical_device_external_memory_host_properties,  // pNext;
+        {},                                                // properties
+    };
+    application->instance()->vkGetPhysicalDeviceProperties2KHR(
+        application->device().physical_device(), &physical_device_properties2);
+
+    if (physical_device_external_memory_host_properties
+            .minImportedHostPointerAlignment > 0) {
+      allocation_size_ = vulkan::RoundUp(
+          allocation_size_, physical_device_external_memory_host_properties
+                                .minImportedHostPointerAlignment);
+    }
+
+    host_pointer_ =
+        aligned_alloc(physical_device_external_memory_host_properties
```

Comment:
can we use the C++ version of this (not 100% sure if msvc implements the C11 version)
std::aligned_alloc

Reply:
That would require C++ 17, right? 
I don't think the project right now requires/specifies use of C++ 17, does it?
I can also use malloc and then align it manually.. if that sounds like a better option

Reply:
@AWoloszyn any thoughts on this?

Reply:
If this builds on Windows,  then I am happy with it.